File: src/main/java/org/evertones/persistence/modules/client/ClientService.java
package org.evertones.persistence.modules.client;

import org.evertones.model.modules.client.Client;
import org.evertones.model.modules.client.ClientDetails;
import org.evertones.model.modules.client.QClient;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDate;

@Service
public class ClientService {

    private ClientRepository clientRepository;

    @Autowired
    public void setClientRepository(ClientRepository clientRepository) {
        this.clientRepository = clientRepository;
    }

    public Client save(Client client) {

        if (client.getClient().getCreatedAt() == null) {
            client.getClient().setCreatedAt(LocalDate.now());
        }
        client.getClient().setUpdatedAt(LocalDate.now());

        return clientRepository.save(client);
    }

    public Client findByClientDetails(ClientDetails clientDetails) {
        return clientRepository.findOne(QClient.client1.client.eq(clientDetails));
    }

}

File: src/main/scala/org/evertones/persistence/modules/client/ClientService.scala
package org.evertones.persistence.modules.client

import java.time.LocalDate

import org.evertones.model.modules.client.{Client, ClientDetails, QClient}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ClientService {

    private var clientRepository: ClientRepository = _

    @Autowired
    def setClientRepository(clientRepository: ClientRepository): Unit = {
        this.clientRepository = clientRepository
    }

    def save(client: Client): Client = {

        if (client.getClient.getCreatedAt == null) {
            client.getClient.setCreatedAt(LocalDate.now())
        }
        client.getClient.setUpdatedAt(LocalDate.now())

        clientRepository.save(client)
    }

    def findByClientDetails(clientDetails: ClientDetails): Client = {
        clientRepository.findOne(QClient.client1.client.eq(clientDetails))
    }

}
